specification.md

```diff
@@ -1642,6 +1647,48 @@ In this case the `failOnValidationErrors` default value of `true` is assumed.
 The `dataInputSchema` property validates the [workflow data input](#Workflow-Data-Input). In case of 
 a starting [Event state](#Event-state), it is not used to validate its event payloads.
 
+The `secrets` property allows you to use sensitive information such as passwords, OAuth tokens, ssh keys, etc. inside your
+Workflow expressions. 
+
+It has two possible types, `string` or `array`. 
+If `string` type, it is an URI pointing to a JSON or YAML document
+which contains an array of names of the secrets, for example:
+
+```json
+"secrets": "file://workflowsecrets.json"
+```
+
+If `array` type, it defines an array (of string types) which contains the names of the secrets, for example:
+
+```json
+"secrets": ["MY_PASSWORD", "MY_STORAGE_KEY", "MY_ACCOUNT"]
+```
+
+For more information about Workflow secrets, reference the [Workflow Secrets section](#Workflow-Secrets).
+
+The `globals` property can be used to define Workflow global values
+which are accessible in [Workflow Expressions](#Workflow-Expressions).
+
+It has two possible types, `string` or `object`.
+If `string` type, it is an URI pointing to a JSON or YAML document
+which contains an object of global definitions, for example:
+
+```json
+"globals": "file://workflowglobals.json"
+```
+
+If `object` type, it defines an array which contains the global definitions, for example:
```

Comment:
I'm blanking on what this should say instead right now, but I think with the latest updates this needs to be updated to not say "array" here, right?

Reply:
it was array before. now it was changed to object type with the latest updates

Reply:
Hmmm, sorry my question must not have been very clear.  

I was asking if you still want this to say "it defines an *array* which contains the global definitions"

Reply:
oops sry good point! will fix

Reply:
fixed